src/swaps/hop/HopSwapProvider.ts

```diff
@@ -0,0 +1,579 @@
+import { Hop } from '@hop-protocol/sdk'
+import { chains, currencyToUnit, unitToCurrency } from '@liquality/cryptoassets'
+import { v4 as uuidv4 } from 'uuid'
+import { createClient } from 'urql'
+import { Wallet } from 'ethers'
+import { SwapProvider } from '../SwapProvider'
+import cryptoassets from '../../utils/cryptoassets'
+import { prettyBalance } from '../../utils/coinFormatter'
+import { withInterval, withLock } from '../../store/actions/performNextAction/utils'
+import { isERC20 } from '../../utils/asset'
+import {
+  BaseSwapProviderConfig,
+  EstimateFeeRequest,
+  EstimateFeeResponse,
+  NextSwapActionRequest,
+  QuoteRequest,
+  SwapRequest,
+  SwapStatus,
+} from '../types';
+import { ActionContext } from '../../store'
+import { Network} from '../../store/types'
+
+export interface HopSwapProviderConfig extends BaseSwapProviderConfig {
+  graphqlBaseURL: string;
+}
+
+class HopSwapProvider extends SwapProvider {
+  config: HopSwapProviderConfig;
+  graphqlURLs: { [key: string]: string };
+
+  constructor(config: HopSwapProviderConfig) {
+    super(config)
+    this.graphqlURLs = {
+      url: this.config.graphqlBaseURL,
+      ethereum: 'hop-mainnet',
+      xdai: 'hop-xdai',
+      arbitrum: 'hop-arbitrum',
+      polygon: 'hop-polygon',
+      optimism: 'hop-optimism'
+    }
+  }
+  /**
+   * Get the supported pairs of this provider for this network
+   * @param {{ network }} network
+   */
+  // eslint-disable-next-line no-unused-vars
+  async getSupportedPairs() {
+    return [];
+  }
+
+  confirmationsSendCount(networkName: string) {
+      const networkToConfirmations: { [key: string]: number } =    
+      {
+        ethereum: 1,
+        arbitrum: 20,
+        polygon: 256
+      }
+      return networkToConfirmations[networkName]
+  }
+
+  gasLimit(networkName: string){
+    const networkToGasLimit : { [key: string]: any } = {
+      arbitrum: {
+        send: 900000,
+        approve: 1000000
+      },
+      polygon: {
+        send: 300000,
+        approve: 300000
+      },
+      ethereum: {
+        send: 150000,
+        approve: 100000
+      }
+    }
+    return networkToGasLimit[networkName]
+  }
+
+  getChain(chainName: string){
+    const slugToChain : { [key: string]: any } = {
+      [Hop.Chain.Ethereum.slug]: Hop.Chain.Ethereum,
+      [Hop.Chain.Arbitrum.slug]: Hop.Chain.Arbitrum,
+      [Hop.Chain.Gnosis.slug]: Hop.Chain.Gnosis,
+      [Hop.Chain.Optimism.slug]: Hop.Chain.Optimism,
+      [Hop.Chain.Polygon.slug]: Hop.Chain.Polygon
+    }
+    return slugToChain[chainName]
+  }
+  // L2->L1 or L2->L2
+  GQL_getDestinationTxHashFromL2Source(transferId: string) {
+    return `query {
+        withdrawalBondeds(
+          where: {
+            transferId: "${transferId}"
+          }
+        ) {
+          timestamp
+          amount
+          transactionHash
+          token
+          timestamp
+        }
+      }
+    `
+  }
+  // L1->L2
+  GQL_getDestinationTxHashFromL1Source(recipient : string) {
```

Comment:
This type of naming seems weird, perhaps have another utility for these, Like what the astroport provider does

Reply:
I moved the queries to separate file